sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -87,6 +96,27 @@
         }));
     }
 
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Instant> renewMessageLock(ServiceBusReceivedMessage messageForLockRenew) {
+        return renewMessageLock(new UUID[]{messageForLockRenew.getLockToken()})
+            .last()
+            .publishOn(scheduler);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Instant> renewMessageLock(UUID lockToken) {
+        return renewMessageLock(new UUID[]{lockToken})
+            .last()
```

Comment:
I don't know if this is the correct operator, what if there are multiple instants? `.single()` would throw an exception if there were more than one element.

Reply:
Normally service would return one because we are sending one lock token . But  in track1 also , they are using same logic since AMQP message comes as array in case of one lock token.

Reply:
that should be fine I guess.